packages/react/tests/components/errorMessage.test.tsx

```diff
@@ -0,0 +1,59 @@
+import { createRef } from 'react'
+import { render, screen } from 'test-utils'
+import { ErrorMessage, FormControlProvider } from '@react'
+
+describe('ErrorMessage', () => {
+  it('renders', () => {
+    render(
+      <FormControlProvider invalid={true}>
+        <ErrorMessage id="test">test message</ErrorMessage>
+      </FormControlProvider>
+    )
+
+    expect(screen.getByText(/test message/i)).toBeInTheDocument()
+  })
+
+  it('forwards refs', () => {
+    const ref = createRef<HTMLParagraphElement>()
+    render(
+      <FormControlProvider invalid={true}>
+        <ErrorMessage id="test" ref={ref}>
+          test message
+        </ErrorMessage>
+      </FormControlProvider>
+    )
+    expect(ref.current).not.toBeNull()
+  })
+
+  it('applies className', () => {
+    render(
+      <FormControlProvider invalid={true}>
+        <ErrorMessage id="test" className="test-class">
+          test message
+        </ErrorMessage>
+      </FormControlProvider>
+    )
+    expect(screen.getByText(/test message/i)).toHaveClass('test-class')
+  })
+
+  it('applies native attributes', () => {
+    const mockFn = jest.fn()
+    render(
+      <FormControlProvider invalid={true}>
+        <ErrorMessage id="test" data-testid="testId" onClick={mockFn}>
+          test message
+        </ErrorMessage>
+      </FormControlProvider>
+    )
+    expect(screen.getByTestId(/testId/i)).toBeInTheDocument()
+    screen.getByText(/test message/i).click()
+    expect(mockFn).toHaveBeenCalled()
+  })
+
+  it("doesn't render when not invalid", () => {
```

Comment:
Would you be okay with changing this to avoid the double negative?  Or do you think that's too different from how the test is performed?

```suggestion
  it("only renders when invalid", () => {
```

Reply:
Oh yeah, that is much better. 😹